src/e2e-test/java/io/cdap/plugin/gcscopy/stepsdesign/GCSCopy.java

```diff
@@ -0,0 +1,158 @@
+  public void enterGCSCopyPropertySourcePath(String path) {
+    GCSCopyActions.enterSourcePath("gs://" + TestSetupHooks.gcsSourceBucketName
+                                     + "/" + PluginPropertyUtils.pluginProp(path));
+  }
+
+  @Then("Enter GCSCopy property destination path")
+  public void enterGCSCopyPropertyDestinationPath() {
+    GCSCopyActions.enterDestinationPath("gs://" + TestSetupHooks.gcsTargetBucketName);
+  }
+
+  @Then("Enter GCSCopy property destination path {string}")
+  public void enterGCSCopyPropertyDestinationPath(String path) {
+    GCSCopyActions.enterDestinationPath("gs://" + TestSetupHooks.gcsTargetBucketName
+                                          + "/" + PluginPropertyUtils.pluginProp(path));
+  }
+
+  @Then("Validate GCSCopy successfully copied object {string} to destination bucket in location {string}")
+  public void validateGCSCopySuccessfullyCopiedObjectToDestinationBucketInLocation(String path, String location) {
+    String targetGCSBucket = TestSetupHooks.gcsTargetBucketName;
+    try {
+      if (StorageClient.getBucketMetadata(targetGCSBucket).getLocation().equalsIgnoreCase(PluginPropertyUtils.pluginProp
+        (location))) {
+        validateGCSCopySuccessfullyCopiedObjectToDestinationBucket(path);
+        return;
+      }
+      Assert.fail("Target gcs bucket " + targetGCSBucket + " is not created in location " + location);
+    } catch (StorageException | IOException e) {
+      if (e.getMessage().contains("The specified bucket does not exist")) {
+        Assert.fail("Target gcs bucket " + targetGCSBucket + " not created - " + e.getMessage());
+      } else {
+        Assert.fail(e.getMessage());
+      }
+    }
+  }
+
+  @Then("Enter runtime argument value {string} for GCSCopy property sourcePath key {string}")
+  public void enterRuntimeArgumentValueForGCSCopyPropertySourcePathKey(String value, String runtimeArgumentKey) {
+    ElementHelper.sendKeys(CdfStudioLocators.runtimeArgsValue(runtimeArgumentKey),
+                           "gs://" + TestSetupHooks.gcsSourceBucketName + "/" + PluginPropertyUtils.pluginProp(value));
+  }
+
+  @Then("Enter runtime argument value for GCSCopy property destination path key {string}")
+  public void enterRuntimeArgumentValueForGCSCopyPropertyDestinationPathKey(String runtimeArgumentKey) {
+    ElementHelper.sendKeys(CdfStudioLocators.runtimeArgsValue(runtimeArgumentKey),
+                           "gs://" + TestSetupHooks.gcsTargetBucketName);
+  }
+
+  @Then("Validate GCSCopy did not copy subdirectory {string} to destination bucket")
+  public void validateGCSCopyDidNotCopySubdirectoryToDestinationBucket(String subdirectory) throws IOException {
+    String sourceGCSBucket = TestSetupHooks.gcsSourceBucketName;
+    String gcsObject = PluginPropertyUtils.pluginProp(subdirectory);
+    for (Blob blob : StorageClient.listObjects(sourceGCSBucket).iterateAll()) {
+      if (blob.getName().contains(gcsObject)) {
```

Comment:
why is this a contains check and not an equals check ?

Reply:
Here it is the scenario to check that the sub-directories are not copied to the destination bucket.This is not working for equals check and the scenario is failing for it when equals check is applied. It shows the assertion error stating that subdirectory is copied from the source GCS bucket.Its working fine with contains check.

Reply:
The comment regarding checking subdirectories in the source bucket (in the method `validateGCSCopySuccessfullyCopiedObjectToDestinationBucket`) also applies here.